src/confd/src/infix-services.c

```diff
@@ -279,6 +292,72 @@ static int restconf_change(sr_session_ctx_t *session, uint32_t sub_id, const cha
 	return put(cfg, srv);
 }
 
+static int ssh_change(sr_session_ctx_t *session, uint32_t sub_id, const char *module,
+		      const char *xpath, sr_event_t event, unsigned request_id, void *_confd)
+{
+	struct lyd_node *ssh = NULL, *listen, *host_key;
+	sr_error_t rc = SR_ERR_OK;
+	sr_data_t *cfg;
+	FILE *fp;
+
+	switch (event) {
+	case SR_EV_DONE:
+		return svc_change(session, event, xpath, "ssh", "sshd");
+	case SR_EV_ENABLED:
+	case SR_EV_CHANGE:
+		break;
+
+	case SR_EV_ABORT:
+	default:
+		return SR_ERR_OK;
+	}
+
+	if (sr_get_data(session, xpath, 0, 0, 0, &cfg) || !cfg) {
+		return SR_ERR_OK;
+	}
+	ssh = cfg->tree;
+
+	if (!lydx_is_enabled(ssh, "enabled")) {
+		goto out;
+	}
+
+	fp = fopen(SSHD_CONFIG_HOSTKEY, "w");
+	if (!fp) {
+		rc = SR_ERR_INTERNAL;
+		goto out;
+	}
+
+	LY_LIST_FOR(lydx_get_child(ssh, "hostkey"), host_key) {
+		const char *keyname = lyd_get_value(host_key);
+		if (!keyname)
+			continue;
+		fprintf(fp, "HostKey %s/hostkeys/%s\n", SSH_BASE, keyname);
+	}
+
+	fclose(fp);
+
+	fp = fopen(SSHD_CONFIG_LISTEN, "w");
+	if (!fp) {
+		rc = SR_ERR_INTERNAL;
+		goto out;
+	}
+
+	LY_LIST_FOR(lydx_get_child(ssh, "listen"), listen) {
+		const char *address, *port;
+		int ipv6;
+		address = lydx_get_cattr(listen, "address");
+		ipv6 = !!strchr(address, ':');
+		port = lydx_get_cattr(listen, "port");
+		fprintf(fp, "ListenAddress %s%s%s:%s\n", ipv6 ? "[" : "", address, ipv6 ? "]" : "", port);
+	}
```

Comment:
Please add some whitespace (empty lines) between variable declarations, and invariants, for readability.  E.g., all if-checks for error handling should be surrounded by newlines.